src/validation/manifest.js

```diff
@@ -111,16 +111,23 @@ const validators = immutable.fromJS({
                       value.get(key),
                       ctx
                         .update('path', path => path.push(key))
-                        .set('type', getFieldType(ctx.get('type'), key))
-                    )
+                        .set('type', getFieldType(ctx.get('type'), key)),
+                    ),
                   )
                 : errors.push(
-                    immutable.fromJS({
-                      path: ctx.get('path'),
-                      message: `Unexpected key in map: ${key}`,
-                    })
+                    key == 'templates'
+                      ? immutable.fromJS({
+                          path: ctx.get('path'),
+                          message: `The way to declare data source templates has changed, ` + 
+                                `please move the templates from inside data sources to ` +
+                                `a \`templates:\` field at the top-level of the manifest.`,
```

Comment:
`top-level` -> `top level`. `top level` is the noun, `top-level` is more like an adjective, I think.